packages/components/src/slot-fill/bubbles-virtually/slot.tsx

```diff
@@ -35,29 +35,33 @@ function Slot(
 		as,
 		// `children` is not allowed. However, if it is passed,
 		// it will be displayed as is, so remove `children`.
-		// @ts-ignore
 		children,
 		...restProps
 	} = props;
 
 	const { registerSlot, unregisterSlot, ...registry } =
 		useContext( SlotFillContext );
+
 	const ref = useRef< HTMLElement >( null );
 
+	// We don't want to unregister and register the slot whenever
+	// `fillProps` change, which would cause the fill to be re-mounted. Instead,
+	// we can just update the slot (see hook below).
+	// For more context, see https://github.com/WordPress/gutenberg/pull/44403#discussion_r994415973
+	const fillPropsRef = useRef( fillProps );
+	useLayoutEffect( () => {
+		fillPropsRef.current = fillProps;
+	}, [ fillProps ] );
```

Comment:
In general this is an indication that something should have been defined using the new `useEvent` maybe.

Reply:
Unfortunately `useEvent` is specialized to support only callbacks. Here we have a value that is a regular object.

Reply:
Yes, if we really wanted to use `useEvent`, it's possible, but it doesn't really improve the code IMO. Also the `fillPropsRef.current` value is used in two effect hooks, which makes `useEvent` even less attractive.